libs/dh/shared/release-toggle/src/lib/dh-release-toggle.directive.spec.ts

```diff
@@ -0,0 +1,448 @@
+})
+class SimpleTestComponent {
+  testId = TEST_IDS.TEST_ELEMENT;
+  expression: ToggleExpression = '';
+}
+
+// Mock service with proper typing
+const mockReleaseToggleService = {
+  [SERVICE_METHODS.IS_ENABLED]: jest.fn(),
+  [SERVICE_METHODS.HAS_ALL_ENABLED]: jest.fn(),
+  [SERVICE_METHODS.REFETCH]: jest.fn(),
+} as jest.Mocked<Pick<DhReleaseToggleService, 'isEnabled' | 'hasAllEnabled' | 'refetch'>>;
+
+describe('DhReleaseToggleDirective', () => {
+  let renderResult: RenderResult<TestHostComponent>;
+
+  const renderComponent = async (componentProperties?: Partial<TestHostComponent>) => {
+    renderResult = await render(TestHostComponent, {
+      imports: [DhReleaseToggleDirective],
+      providers: [
+        { provide: DhReleaseToggleService, useValue: mockReleaseToggleService },
+      ],
+      componentProperties,
+    });
+    return renderResult;
+  };
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    // Reset all mock implementations to default
+    mockReleaseToggleService[SERVICE_METHODS.IS_ENABLED].mockReset();
+    mockReleaseToggleService[SERVICE_METHODS.HAS_ALL_ENABLED].mockReset();
+    mockReleaseToggleService[SERVICE_METHODS.REFETCH].mockReset();
+  });
+
+  describe('Single Toggle Expression', () => {
+    it('should show content when single toggle is enabled', async () => {
+      // Arrange
+      mockReleaseToggleService[SERVICE_METHODS.IS_ENABLED].mockReturnValue(true);
+
+      // Act
+      await renderComponent();
+
+      // Assert
+      const element = screen.getByTestId(TEST_IDS.SINGLE_TOGGLE);
+      expect(element).toBeInTheDocument();
+      expect(element).toHaveTextContent(CONTENT_TEXT.SINGLE_TOGGLE);
+      expect(mockReleaseToggleService[SERVICE_METHODS.IS_ENABLED]).toHaveBeenCalledWith(TOGGLE_NAMES.TOGGLE1);
+    });
+
+    it('should hide content when single toggle is disabled', async () => {
+      // Arrange
+      mockReleaseToggleService[SERVICE_METHODS.IS_ENABLED].mockReturnValue(false);
+
+      // Act
+      await renderComponent();
+
+      // Assert
+      expect(screen.queryByTestId(TEST_IDS.SINGLE_TOGGLE)).not.toBeInTheDocument();
+      expect(mockReleaseToggleService[SERVICE_METHODS.IS_ENABLED]).toHaveBeenCalledWith(TOGGLE_NAMES.TOGGLE1);
+    });
+
+    it('should update visibility when toggle state changes', async () => {
+      // Arrange - Initially disabled
+      mockReleaseToggleService[SERVICE_METHODS.IS_ENABLED].mockReturnValue(false);
+      mockReleaseToggleService[SERVICE_METHODS.HAS_ALL_ENABLED].mockReturnValue(false);
+
+      const { rerender } = await renderComponent();
+      expect(screen.queryByTestId(TEST_IDS.SINGLE_TOGGLE)).not.toBeInTheDocument();
+
+      // Act - Enable toggle and force component to re-evaluate
+      const updatedToggleName = `${TOGGLE_NAMES.TOGGLE1}-updated`;
+      mockReleaseToggleService[SERVICE_METHODS.IS_ENABLED].mockReturnValue(true);
+      rerender({ componentProperties: { singleToggle: updatedToggleName } });
```

Comment:
suggestion: this part - and the same logic in the other tests - is testing whether the content is visible based on a new toggle name. What about keeping the same toggle name but only returning `true`/`false` for the `SERVICE_METHODS.IS_ENABLED` method? This is more like what is happening in reality, doesn't it?